### PR TITLE
Add sourceTransform to Bookmark and MutableImageProperties

### PR DESCRIPTION
**Pull request by tischi**
@K-Meech I started working on it; I will let you know once it is done.

### CODE REVIEW

**Comment by tischi** [comment]
- [ ] modify `showAnnotatedLabelsSource` in imagej-utils to show the source with sourceTransform

**Comment by tischi** [comment]
I think it works now...

**Comment by K-Meech** [comment]
Great! I can take a look tomorrow afternoon

File: src/main/java/de/embl/cba/mobie/image/MutableImageProperties.java [hunk 1]
```diff
@@ -13,4 +13,5 @@
 	public ArrayList< Double > selectedLabelIds;
 	public boolean showSelectedSegmentsIn3d;
 	public boolean showImageIn3d;
+	public double[] addedTransform;
```
Comment (K-Meech) [inline src/main/java/de/embl/cba/mobie/image/MutableImageProperties.java:16]: We should add this to the metadata class too - https://github.com/tischi/imagej-utils/blob/master/src/main/java/de/embl/cba/bdv/utils/sources/Metadata.java

**Review by K-Meech — COMMENTED** [review]
@tischi I added a few comments - I think the main thing is to add sourceTransform to all the Metadata functions too https://github.com/tischi/imagej-utils/blob/master/src/main/java/de/embl/cba/bdv/utils/sources/Metadata.java Thanks!!

File: src/main/java/de/embl/cba/mobie/image/MutableImageProperties.java (hunk 1 above)
Comment (tischi) [reply src/main/java/de/embl/cba/mobie/image/MutableImageProperties.java:16]: This is done in imagej-utils 0.5.10-SNAPSHOT

File: src/main/java/de/embl/cba/mobie/image/ImagePropertiesToMetadataAdapter.java [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
-import net.imglib2.type.numeric.ARGBType;
 import org.apache.commons.lang.WordUtils;
 
 import java.util.ArrayList;
 
-import static de.embl.cba.mobie.utils.Utils.createRandom;
-
 public class ImagePropertiesToMetadataAdapter
 {
 	public void setMetadataFromMutableImageProperties(Metadata metadata, MutableImageProperties imageProperties )
```
[the remark this answers is not in the document]
Comment (tischi) [reply src/main/java/de/embl/cba/mobie/image/ImagePropertiesToMetadataAdapter.java:10]: This is there already (called `addedTransform`), at the bottom of this function.

File: src/main/java/de/embl/cba/mobie/bookmark/BookmarksManager.java [hunk 3]
```diff
@@ -170,15 +165,27 @@ public Bookmark createBookmarkFromCurrentSettings(String bookmarkName) {
...
 		sourcesPanel.updateCurrentMetadata( sourceName );
-		Metadata sourceMetadata = sourcesPanel.getSourceAndCurrentMetadata( sourceName ).metadata();
+		Metadata metadata = sourcesPanel.getSourceAndCurrentMetadata( sourceName ).metadata();
+		Source< ? > source = sourcesPanel.getSourceAndCurrentMetadata( sourceName ).source();
 
 		final ImagePropertiesToMetadataAdapter adapter = new ImagePropertiesToMetadataAdapter();
-		adapter.setMutableImagePropertiesFromMetadata( sourceImageProperties, sourceMetadata );
+		adapter.setMutableImagePropertiesFromMetadata( sourceImageProperties, metadata );
+
+		final int t = 0; // TODO: Once we have data with multiple time points we may have to rethink this...
```
[the remark this answers is not in the document]
Comment (tischi) [reply src/main/java/de/embl/cba/mobie/bookmark/BookmarksManager.java]: Good point! I addressed this, and, along the way, slightly restructured the code.

**Comment by tischi** [comment]
@K-Meech Thanks! Can you have another look?

File: src/main/java/de/embl/cba/mobie/image/MutableImageProperties.java (hunk 1 above)
Comment (K-Meech) [reply src/main/java/de/embl/cba/mobie/image/MutableImageProperties.java:16]: Ok! I don't see it on github though? Did you do it in a branch?

File: src/main/java/de/embl/cba/mobie/bookmark/BookmarksManager.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (K-Meech) [reply src/main/java/de/embl/cba/mobie/bookmark/BookmarksManager.java]: Looks good!

File: src/main/java/de/embl/cba/mobie/image/MutableImageProperties.java (hunk 1 above)
Comment (tischi) [reply src/main/java/de/embl/cba/mobie/image/MutableImageProperties.java:16]: Now?

Comment (K-Meech) [reply src/main/java/de/embl/cba/mobie/image/MutableImageProperties.java:16]: Yes! Can you add it to the copy() function too? https://github.com/tischi/imagej-utils/blob/master/src/main/java/de/embl/cba/bdv/utils/sources/Metadata.java#L90

Comment (tischi) [reply src/main/java/de/embl/cba/mobie/image/MutableImageProperties.java:16]: Done.

**Comment by tischi** [comment]
@K-Meech I merged it. We can fix the rest on the go.

**Pull request merged**